articles/quickstart/native/android/07-authorization.md

```diff
@@ -20,56 +20,58 @@ This step demonstrates how to use Auth0 to create access roles for your users. W
 
 ## Before Starting
 
-Be sure that you have completed the [user profile](04-user-profile) quickstart.
+Be sure that you have completed the [Login](/quickstart/native/android/00-login) quickstart.
 
 ## Create A Rule To Assign Roles
 
-First, you need to create a rule that assigns your users either an `admin` role, or a single `user` role. To do so, go to the [new rule page](${manage_url}/#/rules/new) and select the "*Set Roles To A User*" template, under *Access Control*. Then, replace this line from the default script:
+First, you need to create a rule that assigns your users either an `admin` role, or a simple `user` role. To do so, go to the [new rule page](${manage_url}/#/rules/new) and select the "*Set Roles To A User*" template, under *Access Control*. Then, replace the default script with the contents of the following snippet:
 
-```java
-if (user.email.indexOf('@example.com') > -1)
+```js
+function (user, context, callback) {
+
+  //Define the name of the claim. Must look like a url:
+  //Have 'http' or 'https' scheme and a hostname other than
+  //'auth0.com', 'webtask.io' and 'webtask.run'.
+  var claimName = 'https://access.control/roles';
+
+  //Check if the email has the 'admin.com' domain and give the 'admin' role.
+  //Otherwise, keep a default 'user' role.
+  var roles = ['user'];
+  if (user.email && user.email.indexOf('@admin.com') > -1) {
+      roles.push('admin');
+  }
+  //Set the role claim in the id_token
+  context.idToken[claimName] = roles;
+
+  callback(null, user, context);
+}
 ```
-to match the condition that fits your needs.
 
-By default, it says that if the user email contains `@example.com` then they will be given an `admin` role, otherwise a regular `user` role.
+The rule will run every time a user attempts to authenticate. If the user has a valid `email` and the email's domain is `admin.com`, the user will be granted the `user` and `admin` roles. On any other case, the user will be granted a simple `user` role. This new claim is saved in the *id_token* under the `https://access.control/roles` name, which you will have to check on your app.
 
 ::: note
-You can define more roles other than `admin` and `user`, depending on your product requirements.
+Feel free to customize the required conditions and the different roles to grant on each of them. You can define more roles other than `admin` and `user`, depending on your product requirements. There is a restriction on the naming of the claims, please read [this article](https://auth0.com/docs/rules/current#hello-world) for more context about Rules.
```

Comment:
The link can be shortened here. I might reword this note to be like:

`Feel free to customize the conditions and the roles to grant on each of them. You can define as many roles as your project requires, beyond `admin` and `user`. There are some restrictions on the naming of the claims detailed in the [rules documentation](https://auth0.com/docs/rules/current#hello-world).`

Reply:
And this `https://auth0.com/docs/rules/current#hello-world` should be `/rules#hello-world`